LobbyServer2/LobbyServer/LobbyServerProtocol.cs

```diff
@@ -522,6 +874,80 @@ public void HandleClientErrorSummary(ClientErrorSummary request)
 
         public void HandlePreviousGameInfoRequest(PreviousGameInfoRequest request)
         {
+            // This gets called, when they leave a Custom game creation, not sure if theres anything else that gets called to put this in
+            BridgeServerProtocol customServer = ServerManager.GetCustomServerWithPlayer(AccountId);
+            if (customServer != null)
+            {
+                if (customServer.GameInfo.GameServerProcessCode.StartsWith("CustomGame-"))
+                {
+                    // Remove player from Custom game, if no player left delete the CustomGame itself
+                    BridgeServerProtocol currentServer = ServerManager.GetServerByProcessCode(customServer.GameInfo.GameServerProcessCode);
+
+                    if (currentServer == null)
+                    {
+                        Send(new PreviousGameInfoResponse()
+                        {
+                            ResponseId = request.RequestId,
+                            Success = false
+                        });
+                        // Doesnt Exist anymore
+                        return;
+                    }
+
+                    // Check if Owner Left
+                    if (currentServer.TeamInfo.TeamPlayerInfo.Find(u => u.AccountId == AccountId).IsGameOwner)
+                    {
+                        log.Info($"Owner Left Game kicking all players");
+                        foreach (LobbyServerPlayerInfo playerCheck in currentServer.TeamInfo.TeamPlayerInfo)
+                        {
+                            LobbyServerProtocol playerConnection = SessionManager.GetClientConnection(playerCheck.AccountId);
+                            if (playerCheck.AccountId != AccountId)
+                            {
+                                playerConnection.Send(new GameAssignmentNotification
+                                {
+                                    GameInfo = null,
+                                    GameResult = GameResult.OwnerLeft,
+                                    Reconnection = false
+                                });
+                            }
+                            playerConnection.CurrentServer = null;
+                        }
+
+                        ServerManager.RemoveServer(currentServer.GameInfo.GameServerProcessCode);
```

Comment:
`currentServer.Shutdown()`?

Reply:
It's really confusing that we have fake servers in server manager -- which in fact are just custom game config holders

Reply:
I think it calls for some kind of CustomGameManager

Reply:
I don't see much of the old logic being reused in custom games. If there is a couple of shared fields and methods, you can move them into a common ancestor of BridgeServerProtocol and CustomGame.

It's just bad that you have to keep in mind that this server is not actually a server but a custom game, so you can't call some of its methods or it will explode.